### ISSUE TITLE
On record save JSON data as `jsonBody`

### ISSUE THREAD

**Issue by martin-v** [tomakehurst/wiremock#1399]
`jsonBody` definitely makes the mock files much more readable, but unfortunately the record does not use this feature.
It would be really nice if the recorder persist new mocks with JSON data using this feature.

Probably I can implement this function if it is desired.

**Comment by artmac** [comment]
I agree, because encoded json in as text is not really readable. And manually converting files to jsonBody requires effort which could be avoided.
 
@martin-v have you started to implemented?

**Comment by martin-v** [comment]
Not yet. I wait for a response of the maintainer. No desire to do the wasteful work ;)

@tomakehurst WDYT?

**Comment by tomakehurst** [comment]
I'd be happy to merge a PR implementing this.

Some things to consider:

* What happens if the response from the target server isn't valid JSON? Do we fall back on using a string? Just log a warning? Refuse to record the stub?
* To convert from a string to a JSON object you'll have to deserialise it, which means you'll be subject to however the `ObjectMapper` you're using is configured. This could mean e.g. conversion of nulls to removal of attribute keys.

**Comment by martin-v** [comment]
I would suggest that the set of criteria for using `jsonBody` is:
* The `Content-Type` is one of 
```
application/json
application/x-json
application/ld+json
application/hal+json
application/javascript
application/x-javascript
text/json
text/x-json
text/javascript
text/x-javascript
```

* The size isn't to big, e.g. below 5MB
* The complete response ins valid JSON

Otherwise we fallback zu string.

Anything to add?

**Comment by zufallsgenerator** [comment]
How about adding an option to activate this? While I'd find this very useful, I'd imagine people would be confused if the behaviour suddenly changes with a version upgrade.

**Comment by alexshultsg** [comment]
The work around is to start recording by making POST call to /__admin/recordings/start
with, for example, this
{
  "targetBaseUrl": "<your target>",
  "extractBodyCriteria" : {
    "textSizeThreshold" : "10"
  }
}
and POST /__admin/recordings/stop to stop recording
body file will be saved in __files and mapping witl have bodyFileName field instead of body

**Comment by tomakehurst** [comment]
> I would suggest that the set of criteria for using `jsonBody` is:
> 
> * The `Content-Type` is one of
> 
> ```
> application/json
> application/x-json
> application/ld+json
> application/hal+json
> application/javascript
> application/x-javascript
> text/json
> text/x-json
> text/javascript
> text/x-javascript
> ```
> 
> * The size isn't to big, e.g. below 5MB
> * The complete response ins valid JSON
> 
> Otherwise we fallback zu string.
> 
> Anything to add?

1) I'd suggest just looking for `json` in the right-hand part of the MIME type.
2) Don't include Javascript, as this isn't necessarily valid JSON.

**Comment by steve4835** [comment]
Any update on this?  I would like this feature as well, as it makes reading/editing recorded mappings MUCH more readable
